app/transport/rest/middleware.go

```diff
@@ -2,31 +2,41 @@ package rest
 
 import (
 	"net/http"
+	"time"
 
 	"github.com/go-devs-ua/octagon/lgr"
 )
 
 // Middleware is simple decorator.
-type Middleware func(http.Handler) http.Handler
+type Middleware func(http.Handler, *lgr.Logger) http.Handler
 
 // WrapMiddleware will build middleware chain.
-func WrapMiddleware(h http.Handler, middleware ...Middleware) http.Handler {
+func WrapMiddleware(h http.Handler, logger *lgr.Logger, middleware ...Middleware) http.Handler {
```

Comment:
Use plural form, since it may be more than 1 middleware there.

Reply:
Could you please share references where you learned so good English?  😄

Reply:
No problem 😄 
http://primetime.org.ua/